File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(const vector<int> &nums) {
        int l = 0, r = nums.size() - 1;
        int res = 0;

        if (nums[l] < nums[r]) return nums[l];

        while (l < r) {
            int m = l + (r - l) / 2;

            if (nums[m] > nums[r]) {
                l = m + 1;
            } else {
                r = m;
            }
        }
        return nums[l];
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_sum, max_sub = 0, nums[0]

        for num in nums:
            if current_sum < 0:
                current_sum = 0
            current_sum += num
            max_sub = max(current_sum, max_sub)
        return max_sub